lib/google/gax/api_call/retry_policy.rb

```diff
@@ -0,0 +1,113 @@
+# Copyright 2019, Google Inc.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+#
+#     * Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above
+# copyright notice, this list of conditions and the following disclaimer
+# in the documentation and/or other materials provided with the
+# distribution.
+#     * Neither the name of Google Inc. nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+module Google
+  module Gax
+    class ApiCall
+      ##
+      # The policy for retrying failed API calls using an incremental backoff.
+      # A new object instance should be used for every ApiCall invocation.
+      #
+      class RetryPolicy
+        def initialize(retry_codes: nil, initial_delay: nil, multiplier: nil,
+                       max_delay: nil)
+          @retry_codes   = retry_codes
+          @initial_delay = initial_delay
+          @multiplier    = multiplier
+          @max_delay     = max_delay
+        end
+
+        def retry_codes
+          @retry_codes || [GRPC::Core::StatusCodes::UNAVAILABLE]
```

Comment:
I think the default value for this should be empty. It's a bit counter-intuitive but `UNAVAILABLE` is not safe to retry on without additional information. It's possible for the server to have done some processing but still return an `UNAVAILABLE` status.

Reply:
I’ll make the default be empty. What about the other defaults? Are they okay?

Reply:
Yeah, I think the others are fine.

Reply:
Removed UNAVAILABLE.